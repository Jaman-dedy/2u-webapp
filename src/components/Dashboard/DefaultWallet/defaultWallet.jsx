import React from 'react';
import PropTypes from 'prop-types';
import Message from 'components/common/Message';
import LoaderComponent from 'components/common/Loader';
import MoneySegment from 'components/common/MoneySegment';

const DefaultWallet = ({
  data: { data, loading, error },
  refreshWallet,
}) => {
  return (
    <div>
      <div className="overviewcard">
        <p className="sub-title">My default wallet</p>

        {error && !loading && (
          <Message
            message={
              error.error.error
                ? error.error.error
                : 'Something went wrong'
            }
            action={{
              onClick: () => {
                refreshWallet();
              },
            }}
          />
        )}

        {!error && !loading && data && (
          <MoneySegment
            data={{
              Flag: data.CurrencyFlag,
              Currency: data.Currency,
              Balance: data.Balance,
              Language: data.Language,
            }}
          />
        )}
        {loading && (
          <LoaderComponent loaderContent="Checking Default Wallet Information" />
        )}
      </div>
    </div>
  );
};

DefaultWallet.propTypes = {
  refreshWallet: PropTypes.func,
  data: PropTypes.objectOf(PropTypes.any).isRequired,
};
DefaultWallet.defaultProps = {
  refreshWallet: () => {},
};

export default DefaultWallet;
