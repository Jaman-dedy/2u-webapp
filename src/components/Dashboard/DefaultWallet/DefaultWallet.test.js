import React from 'react';
import { shallow } from 'enzyme';
import DefaultWallet from './defaultWallet';

describe('Default Wallet', () => {
  it('should render correctly', () => {
    const props = {
      setShowWallet: jest.fn(),
      showWallet: false,
      data: {
        loading: false,
        data: [],
      },
    };
    const wrapper = shallow(<DefaultWallet {...props} />);
    expect(wrapper).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
