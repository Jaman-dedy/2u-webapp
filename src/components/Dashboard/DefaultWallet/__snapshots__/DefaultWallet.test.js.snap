// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Default Wallet should render correctly 1`] = `ShallowWrapper {}`;
